src/python/pants/backend/terraform/goals/check.py

```diff
@@ -21,11 +32,41 @@ class TerraformValidateSubsystem(Subsystem):
     skip = SkipOption("check")
 
 
+class SkipTerraformValidateField(BoolField):
+    alias = "skip_terraform_validate"
+    default = False
+    help = "If true, don't run `terraform validate` on this target's code. If this target is a module, `terraform validate might still be run on a `terraform_deployment that references this module."
+
+
+@dataclass(frozen=True)
+class TerraformValidateFieldSet(TerraformFieldSet):
```

Comment:
Doesn't this need to extend the parent's `required_fields` to include `SkipTerraformValidateField`?

Reply:
I didn't know the answer, so I dug around. Seems the typical way does not add the skip field to the required fields, but also doesn't inherit from the parent FieldSet. Usually the fieldset has just the sources field as required. I think that's to let them cover all the target types that have those sources (for example, isort needs to cover `python_source` and `python_test`). The Helm backend, OTOH, subclasses. I think that allows us to avoid duplicating all the fields. I think it makes sense for us to follow the Helm backend, since Terraform files are mostly single-purpose. We might want to revisit that if we decide we want the check both modules and deployments (like helm.kubeconform).

I think using `tgt.get` is what allows the field to be optional and fill in the default.